### DISCUSSION TITLE
How to trigger a build from local repository

### DISCUSSION THREAD

**Discussion by krishnaraghavendra** [concourse/concourse#7997] [Help & Support] [answered]
Hi, I am testing a flow. I used docker to setup concourse locally, now i want to test a pipeline by cloning from local repository/ local folder and run a build on the source code. is this possible 

while creating docker image, i mounted my source code to /my-project. concourse will have access to source code, now i have to run a task on this source code . 
 
- name: concourse-docs-git
  type: git
  icon: github
  source:
    uri: file:///my-project

**Comment by taylorsilva** [comment, marked as answer]
You can't set a pipeline that refers to local files. Concourse wants you to externalize all inputs and access them through resources.

That said, you can kinda do this with the `fly execute` command. You can run a single [Task config](https://concourse-ci.org/tasks.html#schema.task-config) with inputs coming from your local files and/or from a particular job in your pipeline. I'd suggest exploring that command `fly execute --help`

**Reply by sclu1034** [reply]
> Concourse wants you to externalize all inputs and access them through resources.

Is that set in stone?

Because the existence of `fly execute` and the wording of multiple sections in the [tasks doc](https://concourse-ci.org/tasks.html) suggest that you also urge users to externalize task config and task scripts from a `pipeline.yml` into separate files, where you then have to make sure they are available to and in sync with the pipeline.
However, to me, having to maintain a very long `pipeline.yml` looks like much less work than having to maintain an entire separate service to host your task and script files, especially if they are to be kept private, so you either need to handle credentials or implement private network access.

Will there ever be a plan to allow uploading local files as part of `fly set-pipeline`, even if heavily restricted to just task config and scripts?
